pkg/control/pubcontrol/pub_control_utils.go

```diff
@@ -251,9 +253,29 @@ func IsReferenceEqual(ref1, ref2 *policyv1alpha1.TargetReference) bool {
 
 func isNeedPubProtection(pub *policyv1alpha1.PodUnavailableBudget, operation policyv1alpha1.PubOperation) bool {
 	operationValue, ok := pub.Annotations[policyv1alpha1.PubProtectOperationAnnotation]
+	enableInPlacePodVerticalScaling := feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling)
 	if !ok || operationValue == "" {
+		// when operation == RESIZE
+		// 1. if featureGate InPlacePodVerticalScaling is enabled, then do not protect resize
+		// 2. if featureGate InPlacePodVerticalScaling is disabled, then RESIZE is regarded as UPDATE, so return true
+		if operation == policyv1alpha1.PubResizeOperation {
+			return !enableInPlacePodVerticalScaling
+		}
 		return true
 	}
-	operations := sets.NewString(strings.Split(operationValue, ",")...)
+
+	operations := func() sets.Set[string] {
+		operations := sets.New(strings.Split(operationValue, ",")...)
+		// if resize set in pub protect operation, then protect update and resize
+		if operations.Has(string(policyv1alpha1.PubResizeOperation)) {
+			operations.Insert(string(policyv1alpha1.PubUpdateOperation))
+		}
+		// if disable featureGate InPlacePodVerticalScaling, update will contain resize
+		if !enableInPlacePodVerticalScaling && operations.Has(string(policyv1alpha1.PubUpdateOperation)) {
```

Comment:
if disable featureGate InPlacePodVerticalScaling, resize operation will be treated as update operation, so just checking whether update operation should be protected is enough, it seems not necessary to insert extra resize operation in the allowed list.